aws/resource_aws_route53_resolver_dnssec_config.go

```diff
@@ -110,22 +111,80 @@ func resourceAwsRoute53ResolverDnssecConfigRead(d *schema.ResourceData, meta int
 func resourceAwsRoute53ResolverDnssecConfigDelete(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).route53resolverconn
 
-	log.Printf("[DEBUG] Deleting Route53 Resolver DNSSEC config: %s", d.Id())
-	_, err := conn.UpdateResolverDnssecConfig(&route53resolver.UpdateResolverDnssecConfigInput{
-		ResourceId: aws.String(d.Get("resource_id").(string)),
-		Validation: aws.String(route53resolver.ValidationDisable),
-	})
-	if isAWSErr(err, route53resolver.ErrCodeResourceNotFoundException, "") {
+	// To delete a Route53 ResolverDnssecConfig, it must be:
+	// (1) updated to a "DISABLED" state
+	// (2) updated again to be permanently removed
+	//
+	// To determine how many Updates are required,
+	// we first find the config by ID and proceed as follows:
+
+	config, err := finder.ResolverDnssecConfigByID(conn, d.Id())
```

Comment:
i haven't seen much usage of a `Get` method here in the resource _Delete_, but might be appropriate given the Route53 Resolver API's version of "delete" for this particular resource is a 2-step sequence of "Updates". or in this case is it sufficient to just have the resource achieve the "disabled" state? and handle as needed in our sweepers?

Reply:
Some other resources rely on existing attribute state during deletion, which is used to validate the planned assumptions, e.g.

- Resource is disabled and that state is saved
- Resource is planned for destroy
- (Outside Terraform) Resource is re-enabled
- Resource destroy occurs

In this case, the resource would skip the disable step (due to the planned state) and properly return the error about needing to disable first.

Another way to avoid read on delete is to react to error messages similar to the `aws_cloudfront_distribution` resource. In that case, it automatically disables the distribution if it receives the error message about needing to be disabled first.

Using the planned attribute state is probably what these resources _should_ do in most cases, but it is hard to enforce across the provider.